### PR TITLE
feat: validate subscription configuration

### PR DESCRIPTION
**Pull request by ytvnr**
## Issue

https://graviteecommunity.atlassian.net/browse/APIM-138

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/apim-138-validate-subscription-config-2/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-efthakekwg.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/UpdateSubscriptionEntity.java [hunk 1]
```diff
@@ -16,12 +16,22 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
 
 /**
  * @author David BRASSELY (david.brassely at graviteesource.com)
  * @author GraviteeSource Team
  */
+@NoArgsConstructor
+@Getter
+@EqualsAndHashCode(of = "id")
```
Comment (marcambier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/UpdateSubscriptionEntity.java]: equals is not used at all, but was already present, so OK to migrate it to lombok.
It's not very important, but `of` is marked as `Will soon be marked @Deprecated; use the @EqualsAndHashCode.Include annotation together with @EqualsAndHashCode(onlyExplicitlyIncluded = true).`

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java [hunk 2]
```diff
@@ -567,7 +574,12 @@ public SubscriptionEntity update(
                 .orElseThrow(() -> new SubscriptionNotFoundException(updateSubscription.getId()));
 
             if (subscription.getStatus() == Subscription.Status.ACCEPTED) {
+                if (updateSubscription.getConfiguration() != null && !updateSubscription.getConfiguration().isEmpty()) {
+                    subscriptionValidationService.validateAndSanitize(updateSubscription);
+                }
+
                 Subscription previousSubscription = new Subscription(subscription);
+                subscription.setConfiguration(updateSubscription.getConfiguration());
```
Comment (marcambier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java:582]: should we also update filter and metadata ? As it's part of UpdateSubscriptionEntity

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/SubscriptionValidationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    @Override
+    public void validateAndSanitize(UpdateSubscriptionEntity subscription) {
+        subscription.setConfiguration(validateAndSanitizeSubscriptionConfiguration(subscription.getConfiguration()));
+    }
+
+    private String validateAndSanitizeSubscriptionConfiguration(String configuration) {
+        try {
+            String connectorId = objectMapper.readTree(configuration).path("entrypointId").asText();
```
Comment (marcambier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/SubscriptionValidationServiceImpl.java:58]: I'm not very pleased with this duplicated deserialization using the entrypointId magic string (here and in DefaultSubscriptionDispatcher).
But there is no immediate elegant way to do that at all

**Review by marcambier — APPROVED** [review]
Great job, LGTM.
Some minor remarks, but it's ok to me

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java (hunk 2 above)
Comment (ytvnr) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java:582]: good catch, it's done

File: gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/UpdateSubscriptionEntity.java (hunk 1 above)
Comment (ytvnr) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/UpdateSubscriptionEntity.java]: good to know, it's done !

**Pull request merged**